src/posix/platform/multicast_routing.cpp

```diff
@@ -0,0 +1,570 @@
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "posix/platform/multicast_routing.hpp"
+
+#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+
+#include <assert.h>
+#include <chrono>
+#include <net/if.h>
+#include <netinet/icmp6.h>
+#include <netinet/in.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <unistd.h>
+#if __linux__
+#include <linux/mroute6.h>
+#else
+#error "Multicast Routing feature is not ported to non-Linux platforms yet."
+#endif
+
+#include <openthread/backbone_router_ftd.h>
+
+#include "core/common/logging.hpp"
+
+namespace ot {
+namespace Posix {
+
+void MulticastRoutingManager::Init(otInstance *aInstance)
+{
+    otBackboneRouterSetMulticastListenerCallback(aInstance,
+                                                 &MulticastRoutingManager::HandleBackboneMulticastListenerEvent, this);
+}
+
+void MulticastRoutingManager::HandleBackboneMulticastListenerEvent(void *                                 aContext,
+                                                                   otBackboneRouterMulticastListenerEvent aEvent,
+                                                                   const otIp6Address *                   aAddress)
+{
+    static_cast<MulticastRoutingManager *>(aContext)->HandleBackboneMulticastListenerEvent(
+        aEvent, static_cast<const Ip6::Address &>(*aAddress));
+}
+
+void MulticastRoutingManager::HandleBackboneMulticastListenerEvent(otBackboneRouterMulticastListenerEvent aEvent,
+                                                                   const Ip6::Address &                   aAddress)
+{
+    switch (aEvent)
+    {
+    case OT_BACKBONE_ROUTER_MULTICAST_LISTENER_ADDED:
+        Add(aAddress);
+        break;
+    case OT_BACKBONE_ROUTER_MULTICAST_LISTENER_REMOVED:
+        Remove(aAddress);
+        break;
+    }
+}
+
+void MulticastRoutingManager::Enable(void)
```

Comment:
Bubble up the error?

Reply:
Now that we use VerifyOrDie for `InitMulticastRouterSock`, Enable never fail.